File: CSE3063F22_iteration2_classes/Student.java
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.List;

import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;

public class Student extends Person {

    private String studentId;
    private Transcript transcript;
    private String Semester;
    private boolean scheduleStatus = false;

    public Student(String studentId) throws FileNotFoundException, IOException, ParseException {
        super(name, address);
        this.studentId = studentId;
        this.transcript = new Transcript(studentId);
        
        Object obj = new JSONParser().parse(new FileReader("students/"+studentId+".json"));
        JSONObject jo = (JSONObject) obj;
        String SSemester = (String) jo.get("SSemester");
        setSemester(SSemester);
    }
	
    public void setStatus(boolean status) throws FileNotFoundException, IOException, ParseException{
        scheduleStatus=status;
    }

    public boolean getStatus() {
        return scheduleStatus;
    }


    public String getStudentId() {
        return studentId;
    }
    
    public void setStudentId(String studentId) {
	this.studentId = studentId;
    }
    
    public Transcript getTranscript() {
        return transcript;
    }
	
    public void setTranscript(Transcript transcript) {
	this.transcript = transcript;
    }
   
    public String getSemester() {
		return Semester;
	}

     public void setSemester(String semester) {
		Semester = semester;
	}

    public void ReadName() throws IOException, ParseException{
        JSONParser jsonParser=new JSONParser();
        FileReader reader =new FileReader("students/"+getStudentId()+".json");
        Object obj =jsonParser.parse(reader);
        JSONObject sObject=(JSONObject)obj;
        String fname=(String) sObject.get("FirstName");
        String lname=(String) sObject.get("LastName");

        System.out.println("Full Name is : " + fname + " " + lname);
    }

    public void updateStatus() throws FileNotFoundException, IOException, ParseException{
        Object obj = new JSONParser().parse(new FileReader("students/"+getStudentId()+".json"));
        JSONObject jo = (JSONObject) obj;
        jo.remove("ScheduleStatus");
        jo.put("ScheduleStatus", "Approved");

        FileWriter writer = new FileWriter(("students/"+getStudentId()+".json"), false); //overwrites the content of file
        writer.write(jo.toString());
        writer.close();
    }
    
    public void updateJSON(String courseid) throws FileNotFoundException, IOException, org.json.simple.parser.ParseException{
        System.out.println(courseid);
        Object obj = new JSONParser().parse(new FileReader("courses/"+courseid+".json"));
        JSONObject jo = (JSONObject) obj;
        int value = Integer.parseInt(String.valueOf(jo.get("Seatlimit")));
        String new_value = String.valueOf(value-1);
        
        jo.remove("Seatlimit");
        jo.put("Seatlimit", new_value);

        FileWriter writer = new FileWriter("courses/"+courseid+".json",false);
        writer.write(jo.toString());
        writer.close();
     
    }
    
    public void updateSchedule(Course course) throws FileNotFoundException, IOException, ParseException, org.json.simple.parser.ParseException{
        getTranscript().addCoursetoSchedule(course);
        
        Object obj = new JSONParser().parse(new FileReader("students/"+getStudentId()+".json"));
        JSONObject jo = (JSONObject) obj;
        JSONArray Schedule = (JSONArray) jo.get("Schedule");
        
        Schedule.add(course.getCourseId().toString());
    
        FileWriter writer = new FileWriter(("students/"+getStudentId()+".json"), false); //overwrites the content of file
        writer.write(jo.toString());
        writer.close();
     
        
    }

   
}

File: ITERATION3/OOSD_3RD_ITERATİON_code/Student.py
import json

from Course import Course
from Transcript import Transcript
from Person import Person
class Student(Person,object):
    def __init__(self, student_id: str):
       # super().init(name, address)
        self.__student_id = student_id
        self.__transcript = Transcript(student_id)
        self.__schedule_status=False

        try:
            with open(f"students/{student_id}.json", "r") as f:
             data = json.load(f)
            self.__semester = data["SSemester"]
            self.__name = data["FirstName"]
        except FileNotFoundError:
            print('No student file with this ID')

    def set_status(self, status: bool):
        self.__schedule_status = status

    def get_status(self):
        return bool(self.__schedule_status)

    def get_student_id(self) -> str:
        return self.__student_id

    def get_transcript(self) -> Transcript:
        return self.__transcript

    def getSemester(self) -> str:
        return self.__semester

    def get_name(self):
        with open(f"students/{self.__student_id}.json", "r") as f:
            data = json.load(f)
            first_name = data["FirstName"]
            last_name = data["LastName"]
            print("Full Name is: "+first_name+" "+last_name)

    def update_status(self):
        with open(f"students/{self.__student_id}.json", "r") as f:
            data = json.load(f)

        data.pop("ScheduleStatus")
        data["ScheduleStatus"] = "Approved"

        with open(f"students/{self.__student_id}.json", "w") as f:
            json.dump(data, f)

    def update_schedule(self, course: Course):
        self.__transcript.add_course_to_schedule(course)

        with open(f"students/{self.__student_id}.json", "r") as f:
            data = json.load(f)
        
        schedule = data["Schedule"]
        schedule.append(course.get_course_id())
        data["Schedule"]=schedule

        with open(f"students/{self.__student_id}.json", "w") as f:
            json.dump(data, f)
